src/mlpack/core/cv/metrics/ssim_index_impl.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef MLPACK_CORE_CV_METRICS_SSIMINDEX_IMPL_HPP
+#define MLPACK_CORE_CV_METRICS_SSIMINDEX_IMPL_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+	namespace cv {
+
+		template<typename DataType>
+			double SSIMIndex::Evaluate(DataType const& image,
+					DataType const& reference)
+			{
+				if (image.n_cols != reference.n_cols || image.n_rows != reference.n_rows)
```

Comment:
We could simply use arma::size(image) ! = arma::size(reference) here.

Reply:
Done.